packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/semanticdata/listeners/SemanticDataDeletionParticipant.java

```diff
@@ -0,0 +1,33 @@
+package org.eclipse.sirius.web.application.semanticdata.listeners;
+
+import org.eclipse.sirius.web.application.project.dto.DeleteProjectInput;
+import org.eclipse.sirius.web.domain.boundedcontexts.project.events.ProjectDeletedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.projectsemanticdata.events.ProjectSemanticDataDeletedEvent;
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.SemanticData;
+import org.eclipse.sirius.web.domain.boundedcontexts.semanticdata.services.api.ISemanticDataDeletionService;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.event.TransactionalEventListener;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class SemanticDataDeletionParticipant {
```

Comment:
- Missing Javadoc
- We are using the naming convention participant for some stuff already, I would rather see this concept named `SemanticDataCleaner`